src/__tests__/server/db/actions/GameAction.unit.test.ts

```diff
@@ -0,0 +1,355 @@
+        return await createTheme(theme);
+      }),
+    );
+  });
+
+  afterEach(async () => {
+    await GameModel.deleteMany({});
+    await ThemeModel.deleteMany({});
+    await TagModel.deleteMany({});
+  });
+
+  describe("getSelectedGames", () => {
+    test("[pagination] page out of range: expect exception", async () => {
+      //TODO: Expect an exception or an empty list?
+      const numPages = Math.ceil(NUM_GAMES / RESULTS_PER_PAGE);
+      await expect(getSelectedGames({ page: numPages + 1 })).rejects.toThrow(
+        GameNotFoundException,
+      );
+    });
+    test("[pagination] no repeats in paginated results: expect success", async () => {
+      const numPages = Math.ceil(NUM_GAMES / RESULTS_PER_PAGE);
+      const set = new Set();
+      for (let i = 0; i < numPages; i++) {
+        const games = await getSelectedGames({ page: i + 1 });
+        games.games.forEach((game) => {
+          expect(set.has(game)).toBe(false);
+          set.add(game);
+        });
+      }
+      expect(set.size).toBe(NUM_GAMES);
+    });
+    test("[pagination] number of results in each page: expect success", async () => {
+      const numPages = Math.ceil(NUM_GAMES / RESULTS_PER_PAGE);
+      for (let i = 0; i < numPages; i++) {
+        const games = await getSelectedGames({ page: i + 1 });
+        expect(games.games.length).toBe(
+          i < numPages - 1
+            ? RESULTS_PER_PAGE
+            : NUM_GAMES - RESULTS_PER_PAGE * (numPages - 1),
+        );
+      }
+    });
+    test("[tag] nonexistent tag: expect exception", async () => {
+      //Create a non-existent tag name
+      const tag = faker.string.alphanumeric({ length: 40 });
+      //The alphanumeric numeric string is a list of 40 random characters, while the tags are actual names.
+      //There is almost no change this tag will be present in the games.
+      await expect(getSelectedGames({ page: 1, tags: [tag] })).rejects.toThrow(
+        TagNotFoundException,
+      );
+      //Call the API and expect an exception
+      //
+    });
+    test("[tag] in-out groups: expect success", async () => {
+      //Ensure that the length of the in-group is the same as the one of the in group.
+      const tag = faker.helpers.arrayElement(tagInputs);
+      //Loop through all pages to make a list of all games with that tag
+      let query: GameQuery = {
+        page: 1,
+      };
+      //Filter generated gams
+      if (tag.type == "custom") {
+        query.tags = [tag.name];
+      } else {
+        query.accessibility = [tag.name];
+      }
+
+      let results = await getSelectedGames(query);
+      let games = results.games;
+      const numPages = Math.ceil(results.count / RESULTS_PER_PAGE) - 1; //-1 Since one page has already ben traversed.
+      for (let page = 2; page < 2 + numPages; page++) {
+        query.page = page;
+        const result = await getSelectedGames(query);
+        games = [...games, ...result.games];
+      }
```

Comment:
I'm currently using the first call outside the loop to determine the number of pages to use. Do you know how I might be able to get the number of pages without using that call outside the loop and avoiding errors?

Reply:
You don't necessarily have to make that call since this unit test is **specifically** testing the exact results of `tags`. This means that you can assume your action will correctly give you the correct number of pages, so you can derive this from `Math.ceil(NUM_GAMES / RESULTS_PER_PAGE)`.

Reply:
Removed both preliminary calls and simply used tag.games.length and the appropiate one for themes.